web3.js/test/connection.test.ts

```diff
@@ -2952,6 +2952,61 @@ describe('Connection', function () {
     }
   });
 
+  describe('is blockhash valid', () => {
+    it('is blockhash valid - blockhash only (mock)', async () => {
+      const {blockhash} = await helpers.recentBlockhash({connection});
+
+      await mockRpcResponse({
+        method: 'isBlockhashValid',
+        params: [blockhash],
+        value: true,
+        withContext: true
+      });
+
+      const isValid = await connection.isBlockhashValid(blockhash);
+      expect(isValid.value).to.be.true;
+    });
+
+    it('is blockhash valid - blockhash and min slot (mock)', async () => {
+      const {blockhash} = await helpers.recentBlockhash({connection});
+
+      await mockRpcResponse({
+        method: 'isBlockhashValid',
+        params: [blockhash, {minContextSlot: 2}],
+        value: true,
+        withContext: true
+      });
+
+      const isValid = await connection.isBlockhashValid(blockhash, {minContextSlot: 2});
+      expect(isValid.value).to.be.true;
+    });
+
+    if (process.env.TEST_LIVE) {
+      it('is blockhash valid - blockhash only (live)', async () => {
+        const {blockhash} = await helpers.recentBlockhash({connection});
+        const isValid = await connection.isBlockhashValid(blockhash);
+        expect(isValid.value).to.be.true;
+      });
+
+      it('is blockhash valid - blockhash only and mint slot (live)', async () => {
+        const blockhash = await connection.getLatestBlockhashAndContext('finalized');
+
+        const isValid = await connection.isBlockhashValid(
+          blockhash.value.blockhash,
+          {
+            commitment: 'finalized',
+            minContextSlot: blockhash.context.slot+1
+          });
+        expect(isValid.value).to.be.false;
+      });
```

Comment:
Do you have any advice on how to write a test that both exercises `minContextSlot` _and_ fails, @jstarry? I'm missing the plot here.

Reply:
How about setting it to the max safe integer?

Reply:
Setting to MAX_SAFE_INTEGER gives the same result 🤔 validation result is returning as true

Reply:
I love it when writing tests pays off right away.

Reply:
minContextSlot: blockhash.context.slot+1
why adding 1 (+1) after slot?